src/hashstore/filehashstore.py

```diff
@@ -852,16 +947,41 @@ def delete_metadata(self, pid, format_id=None):
         )
         self._check_string(pid, "pid", "delete_metadata")
         checked_format_id = self._check_arg_format_id(format_id, "delete_metadata")
-
+        # Get the metadata directory path for the given pid
         entity = "metadata"
-        metadata_cid = self._computehash(pid + checked_format_id)
-        self._delete(entity, metadata_cid)
-
-        logging.info(
-            "FileHashStore - delete_metadata: Successfully deleted metadata for pid: %s",
-            pid,
-        )
-        return True
+        metadata_directory = self._computehash(pid)
+        rel_path = "/".join(self._shard(metadata_directory))
+        metadata_rel_path = self._get_store_path("metadata") / rel_path
+        if format_id is None:
+            # Delete all metadata files
```

Comment:
danger! see comment in [hashstore.py](https://github.com/DataONEorg/hashstore/pull/88/files#diff-5ea47ca3edd582004c2f6c957e103545fce63b15b6fe93e4d86494593b0da217R172)